src/runtime.rs

```diff
@@ -0,0 +1,484 @@
+async fn start_process_events<'js>(
+    handler: &Function<'js>,
+    base_url: &str,
+    ctx: &Ctx<'js>,
+) -> rquickjs::Result<()> {
+    let exit_iterations = match get_env(_EXIT_ITERATIONS) {
+        Ok(iterations) => iterations.parse::<i64>().unwrap_or(-1),
+        Err(_) => -1,
+    };
+    let mut iterations = 0;
+
+    let client_empty_body = get_hyper_client::<Empty<Bytes>>();
+    let client_full_body = get_hyper_client::<Full<Bytes>>();
+
+    let mut request_id: Option<String> = None;
+    let mut context: Option<LambdaContext> = None;
+    let mut event: Option<Value> = None;
+    loop {
+        let now = Instant::now();
+
+        if let Err(err) = process_event(
+            ctx,
+            base_url,
+            handler,
+            &mut context,
+            &mut event,
+            &client_empty_body,
+            &client_full_body,
+            &mut request_id,
+        )
+        .await
+        .map_err(|e| CaughtError::from_error(ctx, e))
+        {
+            match context {
+                None => Vm::print_error_and_exit(ctx, err),
+                Some(ref context) => {
+                    let error_uri =
+                        format!("{}/invocation/{}/error", base_url, context.aws_request_id);
+                    if let Err(err) = post_error(
+                        &error_uri,
+                        &err,
+                        request_id.as_ref(),
+                        ctx,
+                        &client_full_body,
+                    )
+                    .await
+                    {
+                        Vm::print_error_and_exit(ctx, CaughtError::from_error(ctx, err))
+                    }
+                }
+            }
+        }
+        if exit_iterations > -1 {
+            if iterations >= exit_iterations - 1 {
+                println!("Done in {} ms", now.elapsed().as_millis());
+                break Ok(());
+            }
+            iterations += 1;
+        }
+        context = None;
+        event = None;
+    }
+}
+
+#[allow(clippy::too_many_arguments)]
+async fn process_event<'js>(
+    ctx: &Ctx<'js>,
+    base_url: &str,
+    handler: &Function<'js>,
+    context_lambda: &mut Option<LambdaContext<'js>>,
+    event_: &mut Option<Value<'js>>,
+    client_empty_body: &HyperClient<Empty<Bytes>>,
+    client_full_body: &HyperClient<Full<Bytes>>,
+    request_id: &mut Option<String>,
+) -> rquickjs::Result<()> {
+    let next_invocation_uri = format!("{base_url}/invocation/next");
+    let NextInvocationResponse { event, context } =
+        next_invocation(client_empty_body, next_invocation_uri.as_str(), ctx).await?;
+    if request_id.is_none() {
+        *request_id = Some(context.aws_request_id.clone())
+    };
+    let js_context = convert_into_js_value(ctx.clone(), context.clone())?;
+
+    let js_bootstrap = ctx.globals().get::<_, Object>("__bootstrap")?;
+    let js_set_request_id = js_bootstrap.get::<_, Function>("setRequestId")?;
```

Comment:
Since we now have the client in rust we can remove this here (and from `vm.rs`) and set the global value instead. We can also move `console::LAMBDA_REQUEST_ID` to `runtime.rs`